drivers/serial/uart_stm32.c

```diff
@@ -395,10 +532,16 @@ static const struct uart_stm32_config uart_stm32_cfg_##name = {		\
 	.pclken = { .bus = DT_UART_STM32_##name##_CLOCK_BUS,	\
 		    .enr = DT_UART_STM32_##name##_CLOCK_BITS	\
 	},								\
-	.baud_rate = DT_UART_STM32_##name##_BAUD_RATE		\
 };									\
 									\
 static struct uart_stm32_data uart_stm32_data_##name = {		\
+	.conf = {							\
+		.baudrate = DT_UART_STM32_##name##_BAUD_RATE,		\
+		.parity = UART_CFG_PARITY_NONE,				\
```

Comment:
Do we actually need these fields?
Why wouldn't we be able to use registers for storage and save some RAM?

Reply:
Agreed. I'll push something new shortly.